### PR TITLE
Adding apartment_0000

### PR DESCRIPTION
**Pull request by ahlamiki**
Adding apartment_0000. This scene uses the UE5 latest features, lumen illumination and it has several assets and textures made from scratch for the development of a high quality apartment. In addition to that, I implemented physics constraints to the articulable parts of the furniture in order to be fully functional as a default scene.

### CODE REVIEW

File: cpp/unreal_projects/SpearSim/Config/DefaultEngine.ini [hunk 1]
```diff
@@ -27,11 +27,11 @@ UnrealEdEngine=/Script/SpearSimEditor.SpearSimEditorUnrealEdEngine
 ; Configure Unreal's physics simulation parameters to achieve a reasonable tradeoff between fidelity and performance.
 ; These parameters can be overridden via our config system, but we leave them here also, because editor sessions might
 ; not reach the point where we load the values from our config system.
-[/Script/Engine.PhysicsSettings]
```
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Config/DefaultEngine.ini:30]: Revert all changes to `[/Script/Engine.PhysicsSettings]` since you probably did not intend to make any changes here.

File: cpp/unreal_projects/SpearSim/Config/DefaultEngine.ini [hunk 2]
```diff
@@ -54,7 +54,7 @@ r.DefaultFeature.AutoExposure.ExtendDefaultLuminanceRange=True
 ; Set the default map and game mode
 [/Script/EngineSettings.GameMapsSettings]
 GameDefaultMap=/Game/Scenes/debug_0000/Maps/debug_0000.debug_0000
-EditorStartupMap=/Game/Scenes/debug_0000/Maps/debug_0000.debug_0000
```
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Config/DefaultEngine.ini:57]: Revert this change. I like having `debug_0000` as the default startup map for internal development. We will make `apartment_0000` the default map for external release builds, but that happens in a separate part of the code.

**Comment by ahlamiki** [comment]
Yes, I still have to do the clean up. I am working on it. Sorry for that.

**Comment by mikeroberts3000** [comment]
Comment here when you'd like me to review again.

**Comment by ahlamiki** [comment]
> Comment here when this PR is ready for me to review again.
I am still checking the physic constraints but the scene has been improved and the clean up is done. You can check it again.

**Comment by mikeroberts3000** [comment]
Here are some outstanding issues that I've already commented on but still are not fixed. Please carefully review all of my comments in this PR to make sure they're all addressed.

- Revert all changes to `[/Script/Engine.PhysicsSettings]` since you probably did not intend to make any changes here.
- Revert this change. I like having `debug_0000` as the default startup map for internal development. We will make `apartment_0000` the default map for external release builds, but that happens in a separate part of the code.
- Don't make any changes to `debug_0000`.

Here are some new comments:

- There are some `MI_Kitchen_*` files outside the dedicated kitchen folder. The kitchen folder should have all the kitchen materials. Honestly, I don't think the kitchen folder adds that much to the organization here, because it doesn't contain very many files, it isn't consistent (e.g., there is no dedicated folder for living room), and all the material instances begin with the name of the object anyway (e.g., MI_KitchenSink_*), so they are already sensibly grouped in the Material_Instances folder without this additional layer of folder hierarchy. I think we should remove the kitchen folder and use your existing naming convention to organize the material instances.
- There are some `M_` materials in the material instances folder.
- Spelling error: SM_Exctractor_doors (also why is there a static mesh in the materials folder?)
- Doesn't match your naming convention: MI_Oven_doorGlass, should be MI_Oven_DoorGlass
- Doesn't match your naming convention: `SM__SM_Extractor_*`, should be `SM_Extractor_*`
- Doesn't match your naming convention: `SM_kitchen_base`, should be `SM_Kitchen_Base` (pick a capitalization convention and stick to it 100% of the time)
- Spelling error: Shelve
- I don't think there should be a dedicated vases folder because there aren't that many vases, and you're you're not consistent about it. `Cabinet_Vase` is not in the `Vases` folder. And there are three tables, just like there are three vases. Why don't tables have a dedicated tables folder?

**Comment by mikeroberts3000** [comment]
Clean up the World Outliner. Use our design doc and the latest `debug_0000` as a reference:

- There are several artificial rect lights on the windows. We have discussed this issue specifically, and you assured me that we are not using these lighting hacks. Why are these rect lights here?
- Mimic the organizational structure for the World Outliner as it is described in our design doc, and as exemplified in `debug_0000`. This will require deciding on a semantic label for each actor. Use NYU40 semantic labels as described here (https://github.com/apple/ml-hypersim/blob/main/code/cpp/tools/scene_annotation_tool/semantic_label_descs.csv).
- `SM_wall_*` aren't doing anything. Why are these actors here?
- Move the spheres and the corresponding `GroupActor` into the `Debug` folder.
- Mimic our approach for representing lights that have static meshes associated with them, as described in our design doc, and as exemplified in `debug_0000`. Represent all `PointLight_CeilingLamp_*` as `PointLightComponents` on the corresponding `StaticMeshActor`. Likewise for all the other lights that have static meshes associated with them. The reasoning for this approach is described in detail in our design doc.
- Mimic the way we're doing outdoor illumination in `debug_0000`.
- The `PhysicsConstraintComponents` are not configured correctly. For example, the spatial location of `Chest2.PhysicsConstraint_01` is not the same as the spatial location of `Chest2.Door_01`, even though this is a requirement in our design doc. For now, it is fine if there are still simulation problems after the `PhysicsConstraintComponents` are configured to adhere to all of our requirements, but I want to make sure that we're adhering to all the existing requirements before investing additional human effort into debugging simulation problems.
- Spelling error: `SpotLight_CelingLamp_DinningRoom`
- Spelling error: `SpotLight_KithecnLight_1`

**Comment by mikeroberts3000** [comment]
Minor nitpick: I think the drain in the sink is backwards. It should be directly under the tap, instead of being close to the edge of the counter.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/FabricLinenAO2K.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/FabricLinenAO2K.uasset:1]: Not all assets adhere to our naming convention.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Fireplace/mI_Fireplace_Glass.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Fireplace/mI_Fireplace_Glass.uasset:1]: Doesn't belong here, doesn't match our naming convention

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Extractor/SM_kitchen_base.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Extractor/SM_kitchen_base.uasset:1]: Doesn't match our naming convention.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/NewMaterial.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/NewMaterial.uasset:1]: Doesn't belong here. Doesn't match our naming convention.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_OvenDrawers_drawer_11.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_OvenDrawers_drawer_11.uasset:1]: Doesn't match our naming convention. Why is `drawer_11` lower-case but `Drawer_08` is upper-case?

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Port_Hole/MI_Ceiling_Light.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Port_Hole/MI_Ceiling_Light.uasset:1]: Doesn't belong here.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Shelve/SM_Prop_01.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Shelve/SM_Prop_01.uasset:1]: This folder is spelled incorrectly. I have commented on this exact issue previously. Please go through every one of my previous comments and make sure it is applied throughout the project.

File: cpp/unreal_projects/SpearSim/SpearSim.uproject [hunk 3]
```diff
@@ -1,44 +1,40 @@
...
-            "Name": "UrdfBot",
-            "Enabled": false
-        },
-        {
-            "Name": "Vehicle",
-            "Enabled": true
-        }
-    ]
-}
+	"FileVersion": 3,
```
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/SpearSim.uproject:2]: I don't want these changes. The main difference to our existing code is that you're using tabs instead of spaces. But our convention is to use spaces. I have commented on this exact issue in our previous meeting. You're also disabling the `SimulationController` which breaks our existing code. Why are you doing this? Please revert this entire file.

**Comment by mikeroberts3000** [comment]
I'll review the project structure in the World Outliner on Monday.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Port_Hole/T_Ceilin_Light_d.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Port_Hole/T_Ceilin_Light_d.uasset:1]: Spelling error.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset:1]: Naming not consistent

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Curtain_01.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Curtain_01.uasset:1]: Naming is not consistent: why is there `Curtain_01` and `Curtains_01`?

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset:1]: Naming is not consistent. Why is there `Table_Living` and `Table_LivingRoom`?

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Curtains/T_Curtain_Height.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Curtains/T_Curtain_Height.uasset:1]: Naming is not consistent. Why `T_Curtain` and `T_Curtains`? Is this intentional? Does one apply to a single curtain, and the other applies to multiple curtains?

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Extractor/T_Extractor_Shelve_d.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Extractor/T_Extractor_Shelve_d.uasset:1]: This is spelled incorrectly. The singular form of shelf is shelf. The plural form of shelf is shelves. Shelve is not an English word. I have commented on this exact spelling mistake multiple times.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Sink/T_KitchenSink_Drawers_d.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Sink/T_KitchenSink_Drawers_d.uasset:1]: Naming is not consistent. Why `KitchenSink` sometimes and `Kitchen_Sink` other times?

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset:1]: I have left it while I was checking the folders, the correct convention is "T_First_Letter_Is_Uppercase", which it is applied on the rest of the files.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Curtain_01.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Curtain_01.uasset:1]: I wrote it as "Curtains_01" and "Curtains_02". Probably after changing the name I didn't fix the redirector so it was remaining the singular name.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset:1]: I renamed it as "Table_LivingRoom" but I didn't fix the redirector so it was remaining the singular name.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Curtains/T_Curtain_Height.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Curtains/T_Curtain_Height.uasset:1]: Is not intentional, the correct one is "T_Curtains", it has been fixed.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Extractor/T_Extractor_Shelve_d.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Extractor/T_Extractor_Shelve_d.uasset:1]: I have fixed it.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset
[the remark this answers is not in the document]
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset:1]: It is already fixed.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Sink/T_KitchenSink_Drawers_d.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Sink/T_KitchenSink_Drawers_d.uasset:1]: The correct one is "Kitchen_Sink", it is fixed already.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/LivingRoom_Lamp/SM_LivingRoom_Lamp.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/LivingRoom_Lamp/SM_LivingRoom_Lamp.uasset:1]: Why do we call this folder `LivingRoom_Lamp` (room name first, object name second) but there is another folder called `Table_LivingRoom` (object name first, object name second)? This naming convention is inconsistent.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Vase_01/SM_Vase_01.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Vase_01/SM_Vase_01.uasset:1]: Why is there a separate folder for Vase_01/Vase_02/Vase_03, but there combined folders for Curtains/Cushions/Pictures/Props. What is the rule for deciding whether or not to do a combined folder for multiple objects?

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset
Comment (mikeroberts3000) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset:1]: If your convention is `Table_LivingRoom` (not `Table_Living`), then let's rename the folder `Ceiling_Living_Lamp` to match your convention.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Ceiling_Living_Lamp/SM_Ceiling_LivingRoom_Lamp.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Ceiling_Living_Lamp/SM_Ceiling_LivingRoom_Lamp.uasset:1]: Since your convention is `Table_LivingRoom` (not `Table_Living`), let's rename the folder `Ceiling_Living_Lamp` to `Ceiling_LivingRoom_Lamp` match your existing convention.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_LivingRoom_Lamp.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_LivingRoom_Lamp.uasset:1]: Why `LivingRoom_Lamp` but `Chair_LivingRoom`?

**Comment by mikeroberts3000** [comment]
I get the following warnings when I build and cook via the command-line on Windows:

```
LogMaterial: Warning: [AssetLog] F:\code\github\spear\cpp\unreal_projects\SpearSim\Content\Scenes\apartment_0000\Materials\Materials\M_Master_Opacity.uasset: Failed to compile Material for platform PCD3D_SM6, Default Material will be used in game.
        (Node TextureSampleParameter2D) Sampler type is Color, should be Linear Color for /Game/Scenes/apartment_0000/Meshes/Clutter/Fireplace/T_Fireplace_orm.T_Fireplace_orm
        (Node TextureSampleParameter2D) Sampler type is Color, should be Linear Color for /Game/Scenes/apartment_0000/Meshes/Clutter/Fireplace/T_Fireplace_orm.T_Fireplace_orm

LogMaterial: Warning: [AssetLog] F:\code\github\spear\cpp\unreal_projects\SpearSim\Content\Scenes\apartment_0000\Materials\Materials\M_Master_Opacity.uasset: Failed to compile Material for platform PCD3D_SM5, Default Material will be used in game.
        (Node TextureSampleParameter2D) Sampler type is Color, should be Linear Color for /Game/Scenes/apartment_0000/Meshes/Clutter/Fireplace/T_Fireplace_orm.T_Fireplace_orm
        (Node TextureSampleParameter2D) Sampler type is Color, should be Linear Color for /Game/Scenes/apartment_0000/Meshes/Clutter/Fireplace/T_Fireplace_orm.T_Fireplace_orm

LogMaterial: Warning: Cooking a material resource (in M_Master_Opacity hierarchy) that doesn't have a valid ShaderMap! Shadermap pointer is null.
LogMaterial: Warning: Cooking a material resource (in M_Master_Opacity hierarchy) that doesn't have a valid ShaderMap! Shadermap pointer is null.

LogUObjectGlobals: Warning: [AssetLog] F:\code\github\spear\cpp\unreal_projects\SpearSim\Content\Scenes\apartment_0000\Materials\Material_Instances\MI_Curtains_01.uasset: Failed to load '/Game/Scenes/apartment_0000/Meshes/Clutter/Curtains/T_Curtain_Height': Can't find file.
LogLinker: Warning: [AssetLog] F:\code\github\spear\cpp\unreal_projects\SpearSim\Content\Scenes\apartment_0000\Materials\Material_Instances\MI_Curtains_01.uasset: VerifyImport: Failed to load package for import object 'Package /Game/Scenes/apartment_0000/Meshes/Clutter/Curtains/T_Curtain_Height'
```

Please make sure that the project builds and cooks without warnings on the command-line.

**Comment by mikeroberts3000** [comment]
Lighting looks weird when I open the latest project in the editor:

Screenshot (27)

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/LivingRoom_Lamp/SM_LivingRoom_Lamp.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/LivingRoom_Lamp/SM_LivingRoom_Lamp.uasset:1]: I am using the naming convention "LivingRoom_Lamp (room name first, object name second)".

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Vase_01/SM_Vase_01.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Vase_01/SM_Vase_01.uasset:1]: When there are two or more assets from the same semantic family (they are similar but the color and shape change) then they will be placed in the same folder. I placed Vase_01/Vase_02/Vase_03 inside a folder called "Vases" for more coherence.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Ceiling_Living_Lamp/SM_Ceiling_LivingRoom_Lamp.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Ceiling_Living_Lamp/SM_Ceiling_LivingRoom_Lamp.uasset:1]: Fixed

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_LivingRoom_Lamp.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_LivingRoom_Lamp.uasset:1]: Fixed to LivingRoom_Chair

**Comment by mikeroberts3000** [comment]
Lighting still looks weird for me in the latest project.

Screenshot (28)

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset
Comment (mikeroberts3000) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset:1]: Now that the rest of the project has been corrected, please correct this convention mismatch.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset
Comment (mikeroberts3000) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset:1]: Please "fix up redirectors" in the editor for all redirectors if you haven't already. And please review your changes in the "Files changed" tab on GitHub as well as in the Unreal Editor so you can catch similar issues in future.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset
[the remark this answers is not in the document]
Comment (mikeroberts3000) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset:1]: No it is not.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Carpet/T_macro_heather.uasset:1]: T_Macro_Heather has been fixed.

**Comment by ahlamiki** [comment]
In my last version of the scene (which I uploaded to sourcetree) I fixed up all the redirectors in apartment_0000 and the file "MI_Table_Living" has been changed to "MI_LivingRoom_Table".
________________________________
De: Mike Roberts ***@***.***>
Enviado: martes, 27 de junio de 2023 23:39
Para: isl-org/spear ***@***.***>
Cc: Ahlam ***@***.***>; Author ***@***.***>
Asunto: Re: [isl-org/spear] Adding apartment_0000 (PR #205)


@mikeroberts3000 commented on this pull request.

________________________________

On cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Materials/Material_Instances/MI_Table_Living.uasset<https://github.com/isl-org/spear/pull/205#discussion_r1244390992>:

Please "fix up redirectors" in the editor for all redirectors if you haven't already. And please review your changes in the "Files changed" tab on GitHub as well as in the Unreal Editor so you can catch similar issues in future.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset
[the remark this answers is not in the document]
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset:1]: All the redirectors have been checked folder by folder and fixed. The naming of "SM_Kitchen_Oven_base"
 has been fixed to "SM_Kitchen_Oven_Base" this morning.

[the remark this answers is not in the document]
Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset:1]: It looks that it is not showing the last updates.

**Comment by ahlamiki** [comment]
> Lighting still looks weird for me in the latest project.
> 
> ![Screenshot (28)](https://user-images.githubusercontent.com/2341965/249287511-dd9592ed-96a8-4405-bd4e-acfa21084edb.p
This issue is related to the HDRI, I'm finding out what happen with this file.

**Comment by mikeroberts3000** [comment]
After reading your latest comments in the `COMMENTS.pdf` file you sent me via Discord, I have the following follow-up comments:

- `Materials/Materials` and `Materials/Material_Instances` are good.
- Let's move the textures in `Meshes/*` directly into `Materials/Textures` with no additional subfolders. It is confusing that textures are currently stored in a folder called `Meshes`. And there are textures that apply to multiple meshes (e.g., `T_Cabinet_Vase_Mirror`), so it is confusing to store a shared texture with a particular mesh (e.g., storing a shared texture with `Cabinet` but not with `Vase` or `Mirror`).

**Comment by mikeroberts3000** [comment]
The following comments apply to the Outliner view:

- Move all the actors in the `Composite` folder into the `Meshes` folder (see below for details).
- In the `Meshes` folder, create subfolders using the following semantic category names and IDs, and move all relevant actors into these subfolders:
  - https://github.com/apple/ml-hypersim/blob/main/code/cpp/tools/scene_annotation_tool/semantic_label_descs.csv
  - These semantic categories aren't perfect, but lots of people in the robotics community are already using them.
  - For example, create subfolders like `01_wall`, `02_floor`, `03_cabinet`, etc.
  - Only create a subfolder if you intend to place at least one actor in it.
- Move `Rendering/Lights/Ceiling_Kitchen_Lights` and `Rendering/Lights/Ceiling_LivingRoom_Lights` to the `Meshes/35_lamp` folder. We want these actors to be stored in `Meshes` because they each have a mesh attached to them, and we want them to be in the `35_lamp` folder because `lamp` is the closest semantic category to an overhead light.
- Move `Rendering/Lights/HDRIBackdrop` to `Rendering/HDRI/HDRIBackdrop`. I don't have a definitive justification for this, but I subjectively believe that HDRI actors deserve their own subfolder.
- There are some lights that cause the word "Preview" to be projected onto the scene in the editor viewport. This happens because their mobility is set to `Static`. We should set the mobility of all lights to `Movable` so they interact correctly with Lumen.
- Otherwise the structure in the Outliner is good.

The following comments apply to the World Settings view:

- There are some non-default values in the World Settings view. We should avoid changing things in World Settings unless we have a good reason.

**Comment by ahlamiki** [comment]
> After reading your latest comments in the `COMMENTS.pdf` file you sent me via Discord, I have the following follow-up comments:
> 
> * `Materials/Materials` and `Materials/Material_Instances` look good to me.
> * Let's move the textures in `Meshes/*` directly into `Materials/Textures` with no additional subfolders. It is confusing that textures are currently stored in a folder called `Meshes`. And there are textures that apply to multiple meshes (e.g., `T_Cabinet_Vase_Mirror`), so it is confusing to store a shared texture with a particular mesh (e.g., storing a shared texture with `Cabinet` but not with `Vase` or `Mirror`).

It has been changed.

**Comment by ahlamiki** [comment]
> The following comments apply to the Outliner view:
> 
> * Move all the actors in the `Composite` folder into the `Meshes` folder (see below for details).
> * In the `Meshes` folder, create subfolders using the following semantic category names and IDs, and move all relevant actors into these subfolders:
>   
>   * https://github.com/apple/ml-hypersim/blob/main/code/cpp/tools/scene_annotation_tool/semantic_label_descs.csv
>   * These semantic categories aren't perfect, but lots of people in the robotics community are already using them.
>   * For example, create subfolders like `01_wall`, `02_floor`, `03_cabinet`, etc.
>   * Only create a subfolder if you intend to place at least one actor in it.
> * Move `Rendering/Lights/Ceiling_Kitchen_Lights` and `Rendering/Lights/Ceiling_LivingRoom_Lights` to the `Meshes/35_lamp` folder. We want these actors to be stored in `Meshes` because they each have a mesh attached to them, and we want them to be in the `35_lamp` folder because `lamp` is the closest semantic category to an overhead light.
> * Move `Rendering/Lights/HDRIBackdrop` to `Rendering/HDRI/HDRIBackdrop`. I don't have a definitive justification for this, but I subjectively believe that HDRI actors deserve their own subfolder.
> * There are some lights that cause the word "Preview" to be projected onto the scene in the editor viewport. This happens because their mobility is set to `Static`. We should set the mobility of all lights to `Movable` so they interact correctly with Lumen.
> * Otherwise the structure in the Outliner is good.
> 
> The following comments apply to the World Settings view:
> 
> * There are some non-default values in the World Settings view. We should avoid changing things in World Settings unless we have a good reason.


After following the semantic category names of each subfolder of "Meshes", should I change the naming structure of each SM from e.g. "Big_Carpet" to "bigcarpet" or "big_carpet" ?

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/HDRI/EnviroDome.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/HDRI/EnviroDome.uasset:1]: This file appears to be a redirector. Please "fix up redirectors" in the editor for all redirectors if you haven't already. I have suggested this workflow multiple times.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset
[the remark this answers is not in the document]
Comment (mikeroberts3000) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven/SM_Kitchen_Oven_base.uasset:1]: No. The GitHub website is displaying your branch correctly. The problem is that this file is named incorrectly in your `ahlam/apartment_0000_scene` branch. This link shows the state of your `ahlam/apartment_0000_scene` branch.

https://github.com/isl-org/spear/tree/ahlam/apartment_0000_scene/cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Meshes/Clutter/Kitchen_Oven

Note that this link matches the "Files changed" view on this PR. Please fix this issue and leave a comment here with a screenshot from the "Files changed" tab (or from the link above) that demonstrates the problem is fixed.

File: cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Blueprints/Custom_HDRI_Backdrop.uasset
Comment (mikeroberts3000) [inline cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Blueprints/Custom_HDRI_Backdrop.uasset:1]: @gsmario @ahlamiki Why is this file necessary? Can't we just modify the properties of the default blueprint type in the Details pane without modifying the underlying engine asset? I thought the engine asset only gets modified if we open it up in the Blueprint editor.

Comment (ahlamiki) [reply cpp/unreal_projects/SpearSim/Content/Scenes/apartment_0000/Blueprints/Custom_HDRI_Backdrop.uasset:1]: This "Custom_HDRI_Backdrop" is a Blueprint class HDRI that we have created to replace the old one, which was modified from the engine. We have created a new one (called "Custom_HDRI_Backdrop") just to make the scene more clean. This file is the same one we find in the Outliner, it's not part of the engine, is a customized HDRI created for this scene. And as you say, we can modify the properties of the default blueprint type in the Details and we don't have to modify the engine asset, it gets modified as you say if we open it up in the blueprint editor.

**Review by mikeroberts3000 — APPROVED** [review]
Looks perfect.

As soon as you're ready, please use the "squash and merge" button to merge this PR with the `ue-52` branch, rather than the "merge" button. Doing so will avoid cluttering up the `ue-52` branch with many small commit messages from your dev branch.

After merging your PR, please delete all your dev branches if you're not using them any more (e.g., `ahlam/Lights`, `ahlam/LightingScene`, `ahlam/apartment_0000`, `ahlam/apartment_0000_scene`, `ahlam/spear_indoor_0000`). Doing so will avoid cluttering up our list of active branches with an ever-increasing number of dev branches, and you can always recover an already-merged-then-deleted dev branch if you ever need to.
